js/src/hooks/useSyncableProductsCalculation.js

```diff
@@ -0,0 +1,62 @@
+/**
+ * External dependencies
+ */
+import { useEffect, useCallback } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { API_NAMESPACE } from '.~/data/constants';
+import useApiFetchCallback from './useApiFetchCallback';
+import useCountdown from './useCountdown';
+
+/**
+ * @typedef {Object} SyncableProductsCalculation
+ * @property {number|null} count The number of syncable products. `null` if it's still in the calculation.
+ * @property {()=>Promise} request The function requesting the start of a calculation.
+ * @property {()=>Promise} retrieve The function retrieving the result of a requested calculation by polling with 5-second timer.
+ */
+
+const getOptions = {
+	path: `${ API_NAMESPACE }/mc/syncable-products-count`,
+};
+const postOptions = {
+	...getOptions,
+	method: 'POST',
+};
+
+/**
+ * A hook for communicating with the calculation of the syncable products and for returning the result.
+ *
+ * If a shop has a large number of products, requesting the result with a single API may encounter
+ * a timeout or out-of-memory problem. Therefore, we use an API to schedule a batch for calculating,
+ * and another one to poll the result.
```

Comment:
Same as https://github.com/woocommerce/google-listings-and-ads/pull/1706/files#r985802794

I think the verb is [pull ](https://www.webopedia.com/definitions/pull)